clients/rancher/client.go

```diff
@@ -262,6 +262,19 @@ func (c *Client) GetDownStreamClusterClient(clusterID string) (dynamic.Interface
 	return dynamic, nil
 }
 
+// GetDownStreamClusterContext is a helper function that creates a wrangler context to interact with a specific cluster.
+func (c *Client) GetDownStreamClusterContext(clusterID string) (*wrangler.Context, error) {
```

Comment:
I think it would make more sense Priya if we made it like the steve client and do it like this. But same concept.

https://github.com/rancher/shepherd/blob/2bcb20d8aa618d9029dce71939f31d4b32b49dbc/clients/rancher/v1/client.go#L128

Reply:
@floatingman @caliskanugur what do you think?

Reply:
I'm trying to think of why @igomez06? 
Some questions that come to mind are:
Why the ProxyDownstream function wasn't implemented for the latest rancher client?  
Is it more robust?

I'm more of the mindset that if it works, why add more complexity?

Reply:
Israel, I agree with Dan. I've tested this and it works fine. So, why can't we just implement it as it is in this PR?

Reply:
It boils down to it being a member function in Wrangler Context rather than it being it in two places in rancher/client does that make sense? It same thing I'm just saying to move it to Wrangler Context instead of here. It's not adding complexity just organizing it better. 

It makes no sense have client.WranglerContext and client.GetDownstream() this is about design. 

If you notice we don't do that with the dynamic client either.

Reply:
Israel, Are you asking the function be simply moved to : https://github.com/rancher/shepherd/blob/main/pkg/wrangler/context.go?

Reply:
yes, so it's member function of the Wrangler context. Because the you implemented we are treating the downstream wrangler context and the local cluster wrangler context differently. Why would we do that?

Reply:
I've moved it to the wrangler context. Could you please re-review? If there are further adjustments needed to align it with how ProxyDownstream is implemented, it would be helpful if the PIT crew could assist, as I'm not very familiar with it.